tests/subresource_api_test.go

```diff
@@ -694,6 +694,101 @@ var _ = Describe("[sig-compute]Subresource Api", decorators.SigCompute, func() {
 			})
 		})
 	})
+
+	When("dry-run option is set it should not change anything", func() {
+		It("[test_id:7530] when starting a VM", func() {
+			By("Creating VM")
+			vm := libvmi.NewVirtualMachine(libvmifact.NewAlpine())
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Performing dry run start")
+			err = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{DryRun: []string{metav1.DryRunAll}})
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).To(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"))
+		})
+
+		DescribeTable("when stopping a VM", func(gracePeriod *int64) {
+			By("Creating VM")
+			vm := libvmi.NewVirtualMachine(libvmifact.NewAlpine())
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to start")
+			Eventually(ThisVMIWith(vm.Namespace, vm.Name), 120*time.Second, 1*time.Second).Should(BeRunning())
+
+			By("Getting current vmi instance")
+			originalVMI, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Getting current vm instance")
+			originalVM, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Performing dry run stop")
+			options := &v1.StopOptions{DryRun: []string{metav1.DryRunAll}}
+			if gracePeriod != nil {
+				options.GracePeriod = gracePeriod
+			}
+			err = virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, options)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking that VM is still running")
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vm.Status.Ready).To(BeTrue())
```

Comment:
I dont think so, it seems to work for VMI objects because it looks for the VMI.status.phase [[1]](https://github.com/kubevirt/kubevirt/blob/main/tests/framework/matcher/phase.go#L18) [[2]](https://github.com/kubevirt/kubevirt/blob/main/tests/framework/matcher/phase.go#L70) [[3] ](https://github.com/kubevirt/kubevirt/blob/main/tests/framework/matcher/phase.go#L143)